arches_lingo/src/arches_lingo/components/generic/LabelEditor.vue

```diff
@@ -0,0 +1,141 @@
+<script setup lang="ts">
+import { defineProps, onMounted, ref } from "vue";
+import { fetchControlledListOptions } from "@/arches_lingo/api.ts";
+
+import NonLocalizedString from "@/arches_lingo/components/generic/NonLocalizedString.vue";
+import ReferenceDatatype from "@/arches_lingo/components/generic/ReferenceDatatype.vue";
+// import ResourceInstanceRelationships from "@/arches_lingo/components/generic/ResourceInstanceRelationships.vue";
+
+import {
+    EDIT,
+    LANGUAGE_CONTROLLED_LIST,
+    LABEL_CONTROLLED_LIST,
+    STATUSES_CONTROLLED_LIST,
+    METATYPES_CONTROLLED_LIST,
+    EVENT_TYPES_CONTROLLED_LIST,
+} from "@/arches_lingo/constants.ts";
+
+import type {
+    AppellativeStatus,
+    ControlledListItem,
+    ControlledListItemResult,
+} from "@/arches_lingo/types.ts";
+
+const props = withDefaults(
+    defineProps<{
+        value?: AppellativeStatus;
+    }>(),
+    {
+        value: () => ({}) as AppellativeStatus,
+    },
+);
+const appellative_status = ref(props.value);
+
+function onUpdate(newValue: string) {
+    console.log(newValue);
+}
+
+async function getOptions(listId: string): Promise<ControlledListItem[]> {
+    const parsed = await fetchControlledListOptions(listId);
+    const options = parsed.items.map(
+        (item: ControlledListItemResult): ControlledListItem => ({
+            list_id: item.list_id,
+            uri: item.uri,
+            labels: item.values,
+        }),
+    );
+    return options;
+}
+
+const languageOptions = ref<ControlledListItem[]>([]);
+const typeOptions = ref<ControlledListItem[]>([]);
+const statusOptions = ref<ControlledListItem[]>([]);
+const metatypeOptions = ref<ControlledListItem[]>([]);
+const eventTypeOptions = ref<ControlledListItem[]>([]);
+
+onMounted(async () => {
+    const [languageOpts, typeOpts, statusOpts, metatypeOpts, eventTypeOpts] =
+        await Promise.all([
```

Comment:
this is fine and I'm not blocking on it, but it still may hammer db depending on what the endpoint is doing. Happy to punt though

Reply:
Yeah, let's measure this before we prematurely optimize -- we can always switch to the endpoint that returns all lists and then filter on the frontend.

Reply:
Switching to that endpoint may be helpful in avoiding hard-coded list id's because there is graph/node info... I just started working with this endpoint because I was wary of passing around the entire CLM

Reply:
sweet!